driver-sync/src/test/functional/com/mongodb/client/AbstractAtlasSearchIndexManagementProseTest.java

```diff
@@ -84,7 +91,29 @@ private static boolean hasAtlasSearchIndexHelperEnabled() {
 
     @BeforeEach
     public void setUp() {
-        client = createMongoClient(getMongoClientSettings());
+        MongoClientSettings mongoClientSettings = getMongoClientSettingsBuilder()
+                .writeConcern(WriteConcern.MAJORITY)
+                .readConcern(ReadConcern.MAJORITY)
+                .addCommandListener(new CommandListener() {
+                    @Override
+                    public void commandStarted(final CommandStartedEvent event) {
+                   /* This test case examines scenarios where the write or read concern is not forwarded to the server
+                    for any Atlas Index Search commands. If a write or read concern is included in the command,
+                    the server will return an error. */
+                        if (isSearchIndexCommand(event)) {
+                            BsonDocument command = event.getCommand();
+                            assertFalse(command.containsKey("writeConcern"));
```

Comment:
I realized that I had noticed this earlier and then neglected to comment on it:  asserting like this in any event listener is likely a no-op since the driver (I think) swallows exceptions thrown by listeners and in any case it may throw on a different thread for reactive tests.

You need to instead set some state in the listener and then assert on that state in the main test code.  There are examples of this in many places in the driver.

Since this PR is already merged, you'll have to address it in a follow-up PR.

Reply:
(This is a good example of where writing a failing test first, TDD style, can catch errors in the test itself)